### DISCUSSION TITLE
v command is not working

### DISCUSSION THREAD

**Discussion by Troll2412** [vlang/v#19486] [Installation Issues]
I just installed V on Ubuntu ("V has been successfully build") but the v command does not work. What could be the issue? Also, in VSCodium I cannot run v scripts... Apparently I need to select a debugger or install one. I do have installed the V language support. Sorry for these question, they seem very basic, but I am a total beginner

**Comment by lv37** [comment]
What happens when you try to run it? Do you get an error?

**Reply by Troll2412** [reply]
The v command in the terminal does not work at all. In Codium, I got a simple saved as abc.v and try to run it vie the <Run> Tab and then <Run without debugging>. Oh - just now I tried debugging and I get the error message that no launch cnofiguration has been provided...

**Reply by lv37** [reply]
What do you see when you try to run v in the terminal? Is it 'command not found'?

**Reply by Troll2412** [reply]
yep

**Reply by lv37** [reply]
Go to where you downloaded v and built it, make sure there's a `v` file there, then run `sudo ./v symlink`

**Reply by Troll2412** [reply]
Awesome! That works. Thank you so much. I can the script by using v run abc.v in the terminal. The only thing is still an error message if I try via the <Run> Tab. Something about the launch.json config and a missing executable. What is my workspace folder? Just the v folder?

**Reply by lv37** [reply]
Do you have [v-analyzer vscode extension](https://marketplace.visualstudio.com/items?itemName=VOSCA.vscode-v-analyzer) installed?

**Reply by Troll2412** [reply]
Took me a while. Did it in the terminal and added it to PATH. But it does not show up in Codium...

**Reply by lv37** [reply]
Does v-analyzer show up in codium installed extensions?

**Reply by Troll2412** [reply]
No it does not

**Reply by lv37** [reply]
open Codium then press `ctrl + p` and paste `ext install VOSCA.vscode-v-analyzer` in the popup

**Reply by Troll2412** [reply]
I found that by Google search actually, tried it, does not work. The extension does not exist, so it seems

**Reply by lv37** [reply]
Did you install vscodium as a flatpak, snap or a deb package?

**Reply by Troll2412** [reply]
Definitely not flatpak. But I am not sure about the other two. I am new to Linux, too, sorry. And it has been a while since I installed codium

**Reply by lv37** [reply]
run `snap list` and look for vscodium. If you see it then it's installed through snap

**Reply by Troll2412** [reply]
Ok, I did use snap. It is version 1.82.2.23257, classic

**Reply by lv37** [reply]
is there a vscodium folder in `~/snap`?

**Reply by Troll2412** [reply]
yes

**Reply by lv37** [reply]
can you list its contents (send `ls` output)

**Reply by Troll2412** [reply]
"345" , "common" and and "current"

**Reply by Troll2412** [reply]
But they are empty. Except for common. That one contains .cache. And in there gdk-pxbuf-loaders.cache

**Reply by lv37** [reply]
Go to `/snap/codium/current/usr/share/codium/resources/app/` and send ls output

**Reply by Troll2412** [reply]
There is nothing in 'current'

**Reply by lv37** [reply]
in `/snap/current` or `~/snap/current`?

**Reply by Troll2412** [reply]
Oh, I got it, sorry, my mistake. There is ....../app/extensions. v-analyzer is not listed

**Reply by lv37** [reply]
now look under `/snap/codium/current/usr/share/codium/`

**Reply by lv37** [reply]
i don't have snap installed that's why i cant look for where stuff is myself sorry

**Reply by Troll2412** [reply]
not there

**Reply by lv37** [reply]
ok now add this file `/snap/codium/current/usr/share/codium/resources/app/product.json` with the following:
```json
{
  "nameShort": "Visual Studio Code",
  "nameLong": "Visual Studio Code",
  "extensionsGallery": {
    "serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
    "cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
    "itemUrl": "https://marketplace.visualstudio.com/items"
  }
}
```

**Reply by lv37** [reply]
then restart codium then try to install v-analyzer

**Reply by Troll2412** [reply]
I tried sudo nano filename, but it is readonly.Cannot save the changed file

**Reply by lv37** [reply]
run this command `sudo echo -ne '{\n  "nameShort": "Visual Studio Code",\n  "nameLong": "Visual Studio Code",\n  "extensionsGallery": {\n    "serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",\n    "cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",\n    "itemUrl": "https://marketplace.visualstudio.com/items"\n  }\n}' > /snap/codium/current/usr/share/codium/resources/app/product.json`

**Reply by Troll2412** [reply]
It says: Read-only file system

**Reply by lv37** [reply]
alr can you just install the deb version and uninstall the snap one?

**Reply by lv37** [reply]
then run `echo -ne '{\n  "nameShort": "Visual Studio Code",\n  "nameLong": "Visual Studio Code",\n  "extensionsGallery": {\n	"serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",\n	"cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",\n	"itemUrl": "https://marketplace.visualstudio.com/items"\n  }\n}' > ~/.config/VSCodium/product.json`

**Reply by Troll2412** [reply]
Ok I'll try that. Listen, thank you so very much for your help and patience... I really have to work on my Linux skills

**Reply by Troll2412** [reply]
I did it. But it still does not work.

**Reply by lv37** [reply]
you uninstalled the snap version right?

**Reply by Troll2412** [reply]
Yes. Snap list does not show it anymore

**Reply by lv37** [reply]
try this command instead `echo -ne '{\n  "nameShort": "Visual Studio Code",\n  "nameLong": "Visual Studio Code",\n  "extensionsGallery": {\n    "serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",\n    "cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",\n    "itemUrl": "https://marketplace.visualstudio.com/items"\n  }\n}' > ~/.config/VSCodium/product.json`

**Reply by Troll2412** [reply]
It worked! Thank you so much. v-analyzer is working now. Is it true that v language extension and v-analyzer conflict each other? There was this warning. So now just v-analyzer is active. And there is still the problem with the launch.json file and the missing executable

**Reply by lv37** [reply]
yeah you have to remove the v language extension and only use v-analyzer

**Reply by lv37** [reply]
the launch.json stuff is for the debugger if you don't need to use a debugger then you can just use the `run workspace` button above the main function of your v program.

![image](https://github.com/vlang/v/assets/132778382/e3ed1e84-7d77-4440-bc94-201e9c7b77b5)

**Reply by Troll2412** [reply]
I am just starting to learn V, so I don't think I need a Debugger right now. It works just fine as it is. Thank you thank you thank you thank you

**Reply by Troll2412** [reply]
I don't really want to annoy you again, but this morning, the config that worked yesterday, did not work anymore. I don't know why or what happened. So I re-installed Vlang and got a fresh VSCode Insiders. I was able to install v-analyzer right away. But it's not working. There are 2 error messages: Request textDocument / semanticTokens / range failed . Client V Language Server: connection to server is erroring. Shutting down server.... I actually can run v scripts in the terminal inside VS with v run script.v

**Reply by lv37** [reply]
try running `v up` then `v-analyzer up`

**Reply by lv37** [reply]
then restart vscode

**Reply by Troll2412** [reply]
I tried that earlier, didn't work. Still doesn't. And btw, the v-analyzer up does only work in a certain directory, which I don't remember

**Reply by lv37** [reply]
run `which v-analyzer` and send the output

**Reply by Troll2412** [reply]
This is the error message:

[Info  - 9:22:31 PM] Indexing /home/chris/.config/v-analyzer/metadata
[Info  - 9:22:31 PM] Indexing /home/chris/v/learn_V
signal 11: segmentation fault
                                                        | 0x560632fa4a3c | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x1a2a3c) 
                                                        | 0x560632fb632c | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x1b432c) 
                                                        | 0x560633027d1f | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x225d1f) 
                                                        | 0x560633027e5b | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x225e5b) 
                                                        | 0x7f375a7ef609 | /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) 
                                                        | 0x7f375a929133 | /lib/x86_64-linux-gnu/libc.so.6(clone+0x43) 
[Error - 9:22:31 PM] Connection to server got closed. Server will not be restarted.

**Reply by lv37** [reply]
Are you using the snap version of vscode insiders?

**Reply by Troll2412** [reply]
Yep. Why is that apparently always a problem? I don't get it

**Reply by lv37** [reply]
It's cause snap sandboxes the application and the sandbox may some times be problematic. Try using the deb version

**Reply by Troll2412** [reply]
The error stays the same. I installed regular VSCode, was able to install v-analyzer successfully and:

[Info  - 9:48:08 PM] 
true
[Info  - 9:48:08 PM] v-analyzer version: 0.0.1-beta.1, OS: linux x64
[Info  - 9:48:08 PM] v-analyzer executable path: /home/chris/.config/v-analyzer/bin/v-analyzer
[Info  - 9:48:08 PM] v-analyzer build with V df3c85e
[Info  - 9:48:08 PM] v-analyzer build at 2023-10-02 21:48:08
[Info  - 9:48:08 PM] Client / Editor: Visual Studio Code 1.82.3 (PID: 21358)
[Info  - 9:48:08 PM] Using global config: /home/chris/.config/v-analyzer/config.toml
[Info  - 9:48:08 PM] Found toolchain candidates:
[Info  - 9:48:08 PM]   /home/chris/v
[Info  - 9:48:08 PM] Using "/home/chris/v" as toolchain
[Info  - 9:48:08 PM] Using "/home/chris/.cache/v-analyzer" as cache dir
[Info  - 9:48:08 PM] Using "/home/chris/.vmodules" as vmodules root
[Info  - 9:48:08 PM] Indexing /home/chris/.vmodules
[Info  - 9:48:08 PM] Indexing /home/chris/v/vlib
[Info  - 9:48:08 PM] Indexing /home/chris/.config/v-analyzer/metadata
[Info  - 9:48:08 PM] Indexing /home/chris/v/learn_V
signal 11: segmentation fault
                                                        | 0x55a1d1ae8a3c | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x1a2a3c) 
                                                        | 0x55a1d1afa32c | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x1b432c) 
                                                        | 0x55a1d1b6bd1f | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x225d1f) 
                                                        | 0x55a1d1b6be5b | /home/chris/.config/v-analyzer/bin/v-analyzer(+0x225e5b) 
                                                        | 0x7f6c51105609 | /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) 
                                                        | 0x7f6c5123f133 | /lib/x86_64-linux-gnu/libc.so.6(clone+0x43) 
[Error - 9:48:09 PM] Connection to server got closed. Server will not be restarted.

**Reply by lv37** [reply]
what are the files in /home/chris/v/learn_V

**Reply by Troll2412** [reply]
.  ..  .editorconfig  .git  .gitattributes  .gitignore  src  v.mod

**Reply by lv37** [reply]
codium worked for you right? Maybe just try to use that again

**Reply by Troll2412** [reply]
Ok I'll try that. Thanks again, sorry for bothering you

**Reply by lv37** [reply]
its ok
